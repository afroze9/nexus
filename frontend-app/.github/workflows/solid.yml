name: Build
on:
  push:
    branches:
      - master
  pull_request:
    types: [opened, synchronize, reopened]
jobs:
  analyze:
    name: SonarCloud
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          
      - name: Install Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 16

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        id: pnpm-install
        with:
          version: 7
          run_install: false

      - name: Get pnpm store directory
        id: pnpm-cache
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT

      - uses: actions/cache@v3
        name: Setup pnpm cache
        with:
          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-

      - name: Install dependencies
        run: pnpm install

      - name: Generate coverage
        run: pnpm run coverage

      - name: SonarCloud Scan
        uses: SonarSource/sonarcloud-github-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.PR_TOKEN }} # Needed to get PR information, if any
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
  
  deploy:
    runs-on: ubuntu-latest
    needs: analyze
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          
      - name: Docker Setup Buildx
        uses: docker/setup-buildx-action@v2.5.0

      - name: Docker Login
        uses: docker/login-action@v2.1.0
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}

      - name: Build and push Docker images
        uses: docker/build-push-action@v4.0.0
        with:
          file: Dockerfile
          tags: ${{ secrets.DOCKERHUB_USERNAME }}/nexus-frontend-app:latest, ${{ secrets.DOCKERHUB_USERNAME }}/nexus-frontend-app:${{ github.run_number }}
          push: true
