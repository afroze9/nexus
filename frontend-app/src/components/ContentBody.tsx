import { Component, lazy } from 'solid-js';
import { Flex, Box } from '@hope-ui/solid';
import { Route, Routes } from '@solidjs/router';
import Home from '../pages/Home';

const Companies = lazy(() => import('../pages/companies/Index'));
const CompanyCreate = lazy(() => import('../pages/companies/Create'));
const CompanyDetails = lazy(() => import('../pages/companies/Details'));

const Projects = lazy(() => import('../pages/projects/Index'));
const ProjectCreate = lazy(() => import('../pages/projects/Create'));
const ProjectDetails = lazy(() => import('../pages/projects/Details'));


export const ContentBody: Component = () => {
  return (
    <Flex flex={1} bg="$accent1">
      <Box css={{ width: '100%' }}>
        <Routes>
          <Route path="/" component={Home} />
          <Route path="/companies" component={Companies} />
          <Route path="/companies/create" component={CompanyCreate} />
          <Route path="/companies/:id/details" component={CompanyDetails} />
          <Route path="/projects" component={Projects} />
          <Route path="/projects/create" component={ProjectCreate} />
          <Route path="/projects/:id/details" component={ProjectDetails} />
        </Routes>
      </Box >
    </Flex >
  );
};
