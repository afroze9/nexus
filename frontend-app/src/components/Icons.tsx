import { Icon } from "@hope-ui/solid";
import { FiEdit2 } from 'solid-icons/fi';
import { AiOutlineDelete } from 'solid-icons/ai'

const IconEdit = () => {
  return <Icon as={FiEdit2} />
}

const IconDelete = () => {
  return <Icon as={AiOutlineDelete} />
}

export { IconEdit, IconDelete }
