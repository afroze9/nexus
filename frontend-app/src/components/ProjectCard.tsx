import { Box, Flex, Center, Button } from "@hope-ui/solid"
import { Link } from "@solidjs/router"
import { Component } from "solid-js"

export type ProjectProps = {
  id: number,
  name: string,
  tasks: number,
}

export const ProjectCard: Component<ProjectProps> = (props: ProjectProps) => {
  return (
    <Box
      width="100%"
      borderWidth="1px"
      borderColor="$neutral6"
      borderRadius="$lg"
      overflow="hidden"
    >
      <Flex>
        <Flex direction="column">
          <Box p="$6">
            <Box fontWeight="$semibold" as="h4" lineHeight="$tight" noOfLines={1}>
              {props.name}
            </Box>
          </Box>
          <Box px="$6" pb="$6">
            <Box fontWeight="$normal" as="h4" lineHeight="$normal" noOfLines={1}>
              Tasks: {props.tasks}
            </Box>
          </Box>
        </Flex>
        <Center ml="auto">
          <Box p="$6">
            <Button as={Link} href={`/projects/${props.id}/details`}>View</Button>
          </Box>
        </Center>
      </Flex>
    </Box>
  )
}
