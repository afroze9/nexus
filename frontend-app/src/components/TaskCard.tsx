import { Box, Flex, Center, Button } from "@hope-ui/solid"
import { Component } from "solid-js"

export type TaskProps = {
  id: number,
  title: string,
  description: string,
  isCompleted: boolean,
  onComplete: (taskId: number) => Promise<void>
}

export const TaskCard: Component<TaskProps> = (props: TaskProps) => {
  return (
    <Box
      width="100%"
      borderWidth="1px"
      borderColor="$neutral6"
      borderRadius="$lg"
      overflow="hidden"
    >
      <Flex>
        <Box p="$6">
          <Box fontWeight="$semibold" as="h4" textDecoration={props.isCompleted ? "line-through" : "none"} lineHeight="$tight" noOfLines={1}>
            {props.title}
          </Box>
          <Box fontWeight="$normal" as="p" textDecoration={props.isCompleted ? "line-through" : "none"} lineHeight="$tight" noOfLines={3}>
            {props.description}
          </Box>
        </Box>
        <Center ml="auto">
          <Box p="$6">
            <Button disabled={props.isCompleted} onClick={() => props.onComplete(props.id)}>Complete</Button>
          </Box>
        </Center>
      </Flex>
    </Box>
  )
}
