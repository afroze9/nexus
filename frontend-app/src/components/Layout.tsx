import { Component } from 'solid-js';
import { Flex } from '@hope-ui/solid';
import { TopNav } from "./TopNav";
import { ContentBody } from "./ContentBody";
import { SideNav } from "./SideNav";

export const Layout: Component = () => {
  return (
    <Flex flexDirection="column" flex={1}>
      <TopNav />
      <Flex direction="row" flex={1} bg="$blackAlpha7"> {/* left to right */}
        <SideNav />
        <ContentBody />
      </Flex>
    </Flex>
  );
};
