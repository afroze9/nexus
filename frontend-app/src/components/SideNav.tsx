import { Component, For } from 'solid-js';
import { Flex, VStack } from '@hope-ui/solid';
import { ApplicationLinks } from '../constants';
import { SideNavLink } from './SideNavLink';
import { Hyperlink } from '../@types';

export interface SideNavLinkProps {
  link: Hyperlink;
}

export const SideNav: Component = () => {
  return (
    <Flex direction="column" bg="$accent10" style={{ width: "250px" }}> {/* top to bottom */}
      <VStack>
        <For each={ApplicationLinks}>
          {(link, i) =>
            <SideNavLink link={link} />
          }
        </For>
      </VStack>
    </Flex>
  );
};
