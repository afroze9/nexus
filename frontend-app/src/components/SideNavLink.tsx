import { JSX } from 'solid-js';
import { Anchor, Box, SystemStyleObject } from '@hope-ui/solid';
import { Link } from '@solidjs/router';
import { SideNavLinkProps } from './SideNav';

const boxStyle: SystemStyleObject = {
  width: '100%',
  padding: '$2',
  _hover: {
    background: '$accent8'
  }
};

const anchorStyle: SystemStyleObject = {
  width: '100%',
  _hover: {
    textDecoration: 'none'
  }
};

export const SideNavLink = (props: SideNavLinkProps): JSX.Element => {
  return (
    <Anchor
      as={Link}
      href={props.link.path}
      color="$accent1"
      css={anchorStyle}
    >
      <Box css={boxStyle}>
        {props.link.name}
      </Box>
    </Anchor>
  );
};
