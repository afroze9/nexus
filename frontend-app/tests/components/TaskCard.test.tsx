import { describe, expect, test } from 'vitest'
import { render } from "@solidjs/testing-library"
import { TaskCard } from "../../src/components/TaskCard";
import { HopeProvider } from '@hope-ui/solid';
import { Router } from '@solidjs/router';

describe("TaskCard", () => {
  const Task = {
    id: 1,
    title: "Task 1",
    description: "Task description",
    isCompleted: false,
    onComplete: async (id: number) => console.log('hello')
  };

  test("renders the Task title", () => {
    const { queryByText } = render(() => <Router><HopeProvider><TaskCard {...Task} /></HopeProvider></Router>);
    expect(queryByText('Task 1')).toBeDefined();
  });

  test("renders the Task description", () => {
    const { queryByText } = render(() => <Router><HopeProvider><TaskCard {...Task} /></HopeProvider></Router>);
    expect(queryByText('Task description')).toBeDefined();
  });

  test('matches snapshot', () => {
    const { container, queryByText } = render(() => <Router><HopeProvider><TaskCard {...Task} /></HopeProvider></Router>);
    expect(container.innerHTML).toMatchSnapshot();
  });

});
