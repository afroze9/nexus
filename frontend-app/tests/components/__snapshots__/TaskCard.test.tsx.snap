// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskCard > matches snapshot 1`] = `"<div class=\\"hope-c-PJLV hope-c-PJLV-iccnQfy-css\\"><div class=\\"hope-flex hope-c-PJLV hope-c-PJLV-irwxpV-css\\"><div class=\\"hope-c-PJLV hope-c-PJLV-iimUhLW-css\\"><h4 class=\\"hope-c-PJLV hope-c-PJLV-ijeSgNg-css\\">Task 1</h4><p class=\\"hope-c-PJLV hope-c-PJLV-iKyNOK-css\\">Task description</p></div><div class=\\"hope-center hope-c-jKOUQW hope-c-PJLV hope-c-PJLV-iiLpPre-css\\"><div class=\\"hope-c-PJLV hope-c-PJLV-iimUhLW-css\\"><button type=\\"button\\" role=\\"button\\" class=\\"hope-button hope-c-ivMHWx hope-c-ivMHWx-kcPQpq-variant-solid hope-c-ivMHWx-kWSPeQ-size-md hope-c-ivMHWx-dzltuD-cv hope-c-PJLV hope-c-PJLV-ijhzIfm-css\\">Complete</button></div></div></div></div>"`;
